content/blog/en/version-7-dojo.md

```diff
@@ -0,0 +1,329 @@
+
+```tsx
+import { DataTemplate, createResource } from '@dojo/framework/core/resource';
+import { fetcher } from './personfetcher';
+import { List } from './List';
+
+const template: DataTemplate = {
+	read: fetcher
+};
+
+const resource = createResource(template);
+
+export default factory(function() {
+	return <List resource={resource} />;
+});
+```
+
+The new data middleware allows widgets to read and set queries on the data resource by providing a simple API that enables a set of Options to get shared between multiple widgets. This approach makes building more complex widgets such as typeaheads or loading indicators much simpler as they need only make calls to the data middleware using the shared Options provided.
+
+```tsx
+import { create, tsx } from '@dojo/framework/core/vdom';
+import { data } from '@dojo/framework/core/middleware/data';
+
+const factory = create({ data });
+
+export const List = factory(function Select({ middleware: { data } }) {
+	const { getOrRead, getOptions } = data();
+
+	const items = getOrRead(getOptions());
+});
+```
+
+For more information, please see the [resource reference guide](https://dojo.io/learn/resources).
+
+## A Safer Way of Testing
+
+As part of Dojo 7 testing in Dojo has been given an overhaul to provide support for new features such as functional children, type safety when working with assertion template the new test renderer and general updates that promote testing best practices.
+
+There are five main components to the new test renderer:
+
+-   `renderer`
+    -   The function used to render widgets in the test environment.
+-   `assertion`
+    -   an assertion builder which can be expected against the test renderer.
+-   `wrap`
+    -   A function that wraps widgets and nodes to be used as a type-safe selector when interacting with assertions and the test renderer.
+-   `ignore`
+    -   A utility function to exclude nodes from an assertion
+-   `compare`
+    -   A custom comparator for node properties that can be used with a wrapped test node
+
+The key concept for the new renderer is using the `wrap` function to create wrapped test nodes and widgets that are used within a tests `assertion`, in place of the real widget. This provides the assertions type information based on the wrapped widget and enables identifying the node or widget in the assertion template structure. The same is true for the test renderer itself that working with properties and children can be done in a type-safe way, using the location of the wrapped test node in the assertion template structure to call properties and resolve functional and name children.
+
+The current harness exists in the `@dojo/framework/testing/harness` directory. It will be supported at least through to Dojo 9, giving time for applications to update their existing tests to use the new test renderer. The `harness` imports will automatically be updated when upgrading your application using the `cli-upgrade-app` Dojo CLI command.
```

Comment:
```suggestion
The current harness exists in the `@dojo/framework/testing/harness` directory. It will be supported through at least version 9 of Dojo, giving time for applications to update their existing tests to use the new test renderer. The `harness` imports will automatically be updated when upgrading your application using the `cli-upgrade-app` Dojo CLI command.
```